File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.cpp
class Solution {
public:
   
    bool isvowel(char ch){
        if(ch=='a' || ch == 'e' || ch=='i' || ch == 'o' || ch=='u' ||
        ch == 'A' || ch == 'E' || ch=='I' ||  ch == 'O' || ch=='U' ){
            return true;
        }
        else{
            return false;
        }
    }    

    string reverseVowels(string s) {
        
        int i=0;
        int j=s.length()-1;
        while(i<j){
            if(!isvowel(s[i])){
                i++;
            }
            else if(!isvowel(s[j])){
                j--;
            }
            else{
                swap(s[i],s[j]);
                i++;
                j--;
            }
        }
        return s;
        
    }
};

File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {
        String vowels = "aeiouAEIOU";
        char[] arr = s.toCharArray();
        int i = 0, j = arr.length - 1;
        while (i < j) {
            while (i < j && vowels.indexOf(arr[i]) == -1) i++;
            while (i < j && vowels.indexOf(arr[j]) == -1) j--;
            char temp = arr[i];
            arr[i++] = arr[j];
            arr[j--] = temp;
        }
        return new String(arr);
    }
}
